### DISCUSSION TITLE
Possibility of setting `init_options` for `rust_analyzer`

### DISCUSSION THREAD

**Discussion by tvsfx** [mrcjkb/rustaceanvim#411] [Q&A] [answered]
As a hack(?) to make range formatting work, I had to set `rustfmt.rangeFormatting.enable = true` both in `init_options` and `settings` in lspconfig. Concretely, my rust_analyzer config looks as follows:
```lua
  rust_analyzer = {
      init_options = {
          -- NOTE: for some reason, this has to be set at startup...
          rustfmt = {
              rangeFormatting = {
                  enable = true,
              },
          },
      },
      settings = {
          ...
          rustfmt = {
              rangeFormatting = {
                  enable = true,
              },
          },
          ... 
     }
}
```

I am looking into migrating to rustaceanvim and can see that there is a `default_settings` option, that allows me to set my current `settings`. However, I am seeing no such equivalent for the `init_options` key. Is there a way to set `init_options` when launching the LSP through rustaceanvim?

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

You can add an `init_options` table to  the `vim.g.rustaceanvim.server` config.

**Reply by tvsfx** [reply]
Oh, I didn't realize that would just work; when I glanced over the code I couldn't find any processing of `init_options`, and since `settings` is renamed to `default_settings` in this table, I assumed that keys weren't just passed through, but I guess other keys are. 

This in combination with https://github.com/mrcjkb/rustaceanvim/discussions/49 should suffice then to port my current rust-analyzer setup; thanks!

**Reply by mrcjkb** [reply]
Yep, it wasn't documented 😅
`default_settings` is just so you can override them with a `.vscode/settings.json` (which wouldn't be possible if you set `settings` directly.
